ckanext/stadtzhtheme/dcat/profiles.py

```diff
@@ -0,0 +1,272 @@
+        if any([
+            self._get_dataset_value(dataset_dict, 'contact_uri'),
+            self._get_dataset_value(dataset_dict, 'contact_name'),
+            self._get_dataset_value(dataset_dict, 'contact_email'),
+            self._get_dataset_value(dataset_dict, 'maintainer'),
+            self._get_dataset_value(dataset_dict, 'maintainer_email'),
+            self._get_dataset_value(dataset_dict, 'author'),
+            self._get_dataset_value(dataset_dict, 'author_email'),
+        ]):
+
+            contact_details = BNode()
+
+            g.add((contact_details, RDF.type, VCARD.Organization))
+            g.add((dataset_node, DCAT.contactPoint, contact_details))
+
+            maintainer_email = self._get_dataset_value(dataset_dict, 'maintainer_email')
+            g.add((contact_details, VCARD.hasEmail, URIRef(maintainer_email)))
+
+            items = [
+                ('contact_name', VCARD.fn, ['maintainer', 'author']),
+            ]
+            self._add_triples_from_dict(dataset_dict, contact_details, items)
+
+        # Tags
+        for tag in dataset_dict.get('tags', []):
+            g.add((dataset_node, DCAT.keyword, Literal(tag['name'])))
+
+        # Resources
+        for resource_dict in dataset_dict.get('resources', []):
+            distribution = BNode()
+
+            g.add((dataset_node, DCAT.distribution, distribution))
+            g.add((distribution, RDF.type, DCAT.Distribution))
+
+            #  Simple values
+            items = [
+                ('name', DCT.title, None),
+                ('description', DCT.description, None),
+                ('state', ADMS.status, None),
+            ]
+
+            self._add_triples_from_dict(resource_dict, distribution, items)
+
+            license_id = self._get_dataset_value(dataset_dict, 'license_id')
+            license_title = self._rights(license_id)
+            g.add((distribution, DCT.rights, Literal(license_title)))
+            g.add((distribution, DCT.license, Literal(license_title)))
+
+            #  Lists
+            items = [
+                ('theme', DCAT.theme, None),
+                ('language', DCT.language, None),
+                ('conforms_to', DCT.conformsTo, None),
+            ]
+            self._add_list_triples_from_dict(resource_dict, distribution, items)
+
+            # Format
+            if '/' in resource_dict.get('format', ''):
+                g.add((distribution, DCAT.mediaType,
+                       Literal(resource_dict['format'])))
+            else:
+                if resource_dict.get('format'):
+                    g.add((distribution, DCT['format'],
+                           Literal(resource_dict['format'])))
+
+                if resource_dict.get('mimetype'):
+                    g.add((distribution, DCAT.mediaType,
+                           Literal(resource_dict['mimetype'])))
+
+            # URL
+            url = resource_dict.get('url')
+            download_url = resource_dict.get('download_url')
```

Comment:
resource_dict never has a download_url field, this only exists in DCAT. To know if it's a download or not, you have to check the format. `xml`, `wms`, `wmts`, `wfs`, `rest` are services, and thus don't need a downloadURL field, everything else is a download, where downloadURL and accessURL is the same.